src/v2/view-builder/components/FactorOptions.js

```diff
@@ -49,8 +55,15 @@ export default ListView.extend({
   itemSelector: '.list-content',
 
   initialize: function () {
+    this.step = this.options.appState.get('currentState').step;
+    this.currentState = this.options.appState.get('currentState');
     this.listenTo(this.collection,'selectFactor', function (data) {
-      this.model.set('factorType', data);
```

Comment:
could be  `this.model.set(this.options.name, data)` if you made such change
```
--- a/src/v2/view-builder/internals/FormInputFactory.js
+++ b/src/v2/view-builder/internals/FormInputFactory.js
@@ -14,6 +14,7 @@ const createFactorTypeView = (opt) => {
   return {
     View: FactorOptions,
     options: {
+      name: opt.name,
       collection: new Collection(optionItems),
     }
   };
```

Reply:
👍